### PR TITLE
feat(sheet): statusbar

### PR DESCRIPTION
**Pull request by yuhongz**

### CODE REVIEW

File: packages/ui-plugin-sheets/src/views/status-bar/CopyableStatisticItem.tsx [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    [key: string]: string;
+};
+
+export const CopyableStatisticItem: React.FC<IStatisticItem> = (item: IStatisticItem) => {
+    const localeService = useDependency(LocaleService);
+    const messageService = useDependency(IMessageService);
+    const formateValue = formatNumber(item.value);
+    const copyToClipboard = async () => {
+        try {
+            await navigator.clipboard.writeText(item.value.toString());
```
Comment (wzhudev) [inline packages/ui-plugin-sheets/src/views/status-bar/CopyableStatisticItem.tsx:33]: Use clipboard interface here.

File: packages/ui-plugin-sheets/src/views/status-bar/StatusBar.tsx [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+                        }
+                        return stat;
+                    });
+                    setStatistics(newStatistics);
+                }
+                setIsSingle(newSingleState);
+            }
+        };
+
+        window.addEventListener('resize', handleResize);
```
Comment (wzhudev) [inline packages/ui-plugin-sheets/src/views/status-bar/StatusBar.tsx:80]: @jikkai The UI plugin should expose a service for components to subscribe to layout changes (not only window resize event).

File: packages/ui-plugin-sheets/src/services/status-bar.service.ts [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+export class StatusBarService implements IStatusBarService, IDisposable {
+    private readonly functions = [
+        FUNCTION_NAMES.SUM,
+        FUNCTION_NAMES.MAX,
+        FUNCTION_NAMES.MIN,
+        FUNCTION_NAMES.AVERAGE,
+        FUNCTION_NAMES.COUNT,
+    ];
+    private readonly _state$ = new BehaviorSubject<Nullable<IStatusBarServiceStatus>>(null);
```
Comment (wzhudev) [inline packages/ui-plugin-sheets/src/services/status-bar.service.ts:27]: IMHO the status should be stored in the local storage and get loaded from the local storage when the web page refreshes.

File: packages/ui-plugin-sheets/src/services/status-bar.service.ts [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    getFunctions(): Readonly<FUNCTION_NAMES[]>;
+}
+
+export type IStatusBarServiceStatus = Array<{
+    func: FUNCTION_NAMES;
+    value: number;
+}>;
+
+export class StatusBarService implements IStatusBarService, IDisposable {
+    private readonly functions = [
```
Comment (wzhudev) [inline packages/ui-plugin-sheets/src/services/status-bar.service.ts:20]: Privde fields' names should start with an underbar.

File: packages/ui-plugin-sheets/src/controllers/status-bar.controller.ts [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+    private _registerSelectionListener(): void {
+        this._selectionManagerService.selectionInfo$.subscribe((selectionRangeWithStyle) => {
+            if (selectionRangeWithStyle) {
+                this._calculateSelection(selectionRangeWithStyle);
+            }
+        });
+    }
+
+    private _calculateSelection(selections: ISelectionWithStyle[]) {
+        const unitId = this._univerInstanceService.getCurrentUniverSheetInstance().getUnitId();
```
Comment (wzhudev) [inline packages/ui-plugin-sheets/src/controllers/status-bar.controller.ts:36]: There are redundant calls to `this._univerInstanceService.getCurrentUniverSheetInstance()`. You only have to call it once.

File: packages/base-formula-engine/src/value-object/primitive-object.ts [hunk 6]
```diff
@@ -204,7 +204,12 @@ export class NumberValueObject extends BaseValueObject {
         if (valueObject.isArray()) {
             return valueObject.plus(this);
         }
-        return this.plusBy(valueObject.getValue());
+        const object = this.plusBy(valueObject.getValue());
```
Comment (wzhudev) [inline packages/base-formula-engine/src/value-object/primitive-object.ts:207]: @DR-Univer Please make sure this is appropriate.

**Review by wzhudev — CHANGES_REQUESTED** [review]
And some interaction problems need to be discussed.

File: packages/base-formula-engine/src/value-object/primitive-object.ts (hunk 6 above)
Comment (yuhongz) [reply packages/base-formula-engine/src/value-object/primitive-object.ts:207]: This part of code was provided by @DR-Univer 🤣

File: packages/ui-plugin-sheets/src/services/status-bar.service.ts (hunk 3 above)
Comment (yuhongz) [reply packages/ui-plugin-sheets/src/services/status-bar.service.ts:27]: The configuration will not be cached locally. Actually, status bar will display three or one statistical item by default based on the current size. Reference -> StatusBar.tsx 😄

**Review by wzhudev — APPROVED** [review]

**Pull request merged**